src/tools/cephfs/cephfs-shell

```diff
@@ -876,16 +876,21 @@ class CephFSShell(Cmd):
                 continue
             else:
                 is_pattern = False
-            path = os.path.normpath(os.path.join(cephfs.getcwd(), path))
+
             if args.parent:
-                files = reversed(sorted(set(dirwalk(path))))
+                path = os.path.join(cephfs.getcwd(), path.rsplit(b'/')[0])
+                files = list(reversed(sorted(set(dirwalk(path)))))
+                if not files:
+                    path = b'.'
                 for filepath in files:
-                    filepath = os.path.normpath(filepath)
-                    if filepath[1:] != path:
-                        try:
-                            cephfs.rmdir(filepath)
-                        except libcephfs.Error:
-                            cephfs.unlink(filepath)
+                    try:
+                        cephfs.rmdir(os.path.normpath(filepath))
+                    except libcephfs.Error as e:
+                        perror(e)
+                        path = b'.'
+                        break
```

Comment:
Wouldn't the right behaviour be too continue the loop instead of break out of it? Breaking out of it would leave rest of the files undeleted.

And if one of the files couldn't be, ideally, the command should attempt to quit right away. Although, this does happen, but it would be better if code does that clearly. So setting a flag in the `execpt` block and then checking the flag and quitting (depending on flag) would be nicer and cleaner, IMO.

Reply:
In unix rmdir this does not happen, it throws error for the invalid file but deletes the rest.
```
[varsha@localhost testdir]$ mkdir -p t1/t2
[varsha@localhost testdir]$ ls
t1
[varsha@localhost testdir]$ rmdir -p inval_dir t1/t2
rmdir: failed to remove 'inval_dir': No such file or directory
[varsha@localhost testdir]$ ls
[varsha@localhost testdir]$ 
```

Reply:
> ```[varsha@localhost testdir]$ rmdir -p inval_dir t1/t2
rmdir: failed to remove 'inval_dir': No such file or directory
[varsha@localhost testdir]$ ls
[varsha@localhost testdir]$ 
```

Umm, I wasn't addressing this case. Anyways, I feel the point I was trying to make falls outside the scope of this PR. I'll go ahead approve. :)

Reply:
Sorry, I did not get your point. Can you please elaborate it again?